core/drivers/gic.c

```diff
@@ -858,6 +860,36 @@ void gic_dump_state(void)
 	}
 }
 
+TEE_Result gic_spi_release_to_ns(size_t it)
+{
+	struct gic_data *gd = &gic_data;
+	size_t idx = it / NUM_INTS_PER_REG;
+	uint32_t mask = BIT32(it % NUM_INTS_PER_REG);
+
+	if (it < gd->max_it && it >= GIC_SPI_BASE)
+		return TEE_ERROR_BAD_PARAMETERS;
+	/* Make sure it's already disabled */
+	if (!gic_it_is_enabled(gd, it))
+		return TEE_ERROR_BAD_STATE;
+	/* Assert it's secure to start with */
+	if (!gic_it_get_group(gd, it))
+		return TEE_ERROR_SECURITY;
+
+	gic_it_set_cpu_mask(gd, it, 0);
+	gic_it_set_prio(gd, it, GIC_SPI_PRI_NS_EL1);
+
+	mutex_lock(&gic_mutex);
```

Comment:
It is not expected this function is called concurrently for the same interrupt, but for consistency I think it would be better the mutex is locked right before line 872, at least before line 878.

Reply:
Done